core/signal.c

```diff
@@ -467,8 +467,7 @@ int uwsgi_signal_wait(struct wsgi_request *wsgi_req, int signum) {
 	return received_signal;
 }
 
-void uwsgi_receive_signal(struct wsgi_request *wsgi_req, int fd, char *name, int id) {
-
+pid_t uwsgi_receive_signal(struct wsgi_request *wsgi_req, int fd, char *name, int id) {
```

Comment:
is this required ? uwsgi_receive_signal is always managed by the process calling it